src/frontend/src/binder/insert.rs

```diff
@@ -166,19 +210,48 @@ impl Binder {
             ))));
         }
 
+        // create table t1 (v1 int, v2 int); insert into t1 (v2) values (5);
+        // We added the null values above. Above is equivalent to
+        // insert into t1 values (NULL, 5);
+        let target_table_col_indices = if !target_table_col_indices.is_empty() && nulls_inserted > 0
+        {
+            let provided_insert_cols: HashSet<usize> =
+                target_table_col_indices.iter().cloned().collect();
+            let all_insert_cols: HashSet<usize> = (0..columns_to_insert.len())
+                .collect::<Vec<usize>>()
+                .iter()
+                .cloned()
+                .collect();
+            let null_cols: HashSet<usize> = (&all_insert_cols - &provided_insert_cols)
+                .iter()
+                .cloned()
+                .collect();
+
+            let mut result: Vec<usize> = target_table_col_indices.clone();
+            for val in null_cols {
+                result.push(val);
+            }
+            result
+        } else {
```

Comment:
Iterate over hashset is non-deterministic. Not sure whether there are problems.

Better to rewrite it to simple loop:

```rust
            for i in 0..columns_to_insert.len() {
                if !provided_insert_cols.contains(&i) {
                    result.push(i);
                }
            }
```

Reply:
I will. Just as a suggestion: We also could have collected to a vector: 

```rust
let null_cols: Vec<usize> = (&all_insert_cols - &provided_insert_cols)
    .iter()
    .cloned()
    .collect();
```